File: GreatOutdoor/GreatOutdoor.Entities/Address.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GreatOutdoor.Entities
{
    //Address  Interface
    public abstract class IAddress
    {
         int AddressID { get; set; }
         string Address_line1 { get; set; }
         string Address_line2 { get; set; }
         string Landmark { get; set; }
         string City { get; set; }
         string State { get; set; }
         int Pincode { get; set; }
         int RetailerID { get; set; }

    }
    //Address Class
    public class Address: IAddress
    {
        

        public int AddressID { get; set; }
        public string Address_line1 { get ; set ; }
        public string Address_line2 { get; set ; }
        public string Landmark { get ; set; }
        public string City { get; set ; }
        public string State { get; set; }
        public int Pincode { get; set ; }
        public int RetailerID { get; set; }
    }
}

File: Prafull Angular/GreatOutdoor/src/app/Models/address.ts
// Address Model Class
export class Address {
  id: number;
  addressID:string;
  addressLine1: string;
  addressLine2: string;
  landmark: string;
  city: string;
  state: string;
  pinCode: string;
  retailerID: string;
  creationDateTime: string;
  lastModifiedDateTime: string;
  //Address Model Class consructor
  constructor(ID: number,AddressID:string, AddressLine1: string, AddressLine2: string, Landmark: string, City: string, State: string,PinCode:string,RetailerID:string, CreationDateTime: string, LastModifiedDateTime: string) {
    this.id = ID;
    this.addressID = AddressID;
    this.addressLine1 = AddressLine1;
    this.addressLine2 = AddressLine2;
    this.landmark = Landmark;
    this.city = City;
    this.state = State;
    this.pinCode = PinCode;
    this.retailerID = RetailerID;
    this.creationDateTime = CreationDateTime;
    this.lastModifiedDateTime = LastModifiedDateTime;
  }
}


